Run stubtest on stubs on different platforms (#8923)
python/typeshed: CONTRIBUTING.md

* `apt_dependencies` (default: `[]`): A list of Ubuntu APT packages
  that need to be installed for stubtest to run successfully. These are
  usually packages needed to pip install the implementation distribution.

> * `platforms` (default: `["linux"]`): A list of supported OSes.
>   Can contain `win32`, `linux`, and `darwin` values.
>   If not specified, only `linux` is considered supported.

Reviewer: ```suggestion
* `platforms` (default: `["linux"]`): A list of OSes on which to run stubtest.
  Can contain `win32`, `linux`, and `darwin` values.
  If not specified, tests are run only on `linux`.
```

>   If not specified, tests are run only on `linux`.

Reviewer 2: ```suggestion
  If not specified, stubtest is run only on `linux`.
```

>   Only add extra OSes to the test
>   if there is platform-specific code in a package.

Author: ```suggestion
  Only add extra OSes to the test
  if there are platform-specific branches in a stubs package.
```

* `apt_dependencies` (default: `[]`): A list of Ubuntu APT packages
  that need to be installed for stubtest to run successfully. These are
  usually packages needed to pip install the implementation distribution.
* `platforms` (default: `["linux"]`): A list of supported OSes.
  Can contain `win32`, `linux`, and `darwin` values.

>   If not specified, only `linux` is considered supported.

Reviewer: We should specify that this should be set only if there's some specific reason to test on multiple OSes, because we don't want useless duplicate stubtest runs for all pure-Python packages.

* `platforms` (default: `["linux"]`): A list of supported OSes.
  Can contain `win32`, `linux`, and `darwin` values.
  If not specified, only `linux` is considered supported.
  Only add extra OSes to the test

>   if there is platform-specific code in a package.

Reviewer 3: Just a suggestion: I'd specify platform-specific stubs. PyAutoGUI is a good example that has platform-specific code, but the public module (the one we stub), has platform-agnostic typing. (hence it doesn't need to be stub-tested on multiple platforms).

vobject, cffi, setuptools, colorama, etc. have specific allowlist entries because of windows, these are example of stubs that would benefit from running on multiple OSes. (less exclusions, more completion)

Reviewer 3: Maybe something around the lines of `if there are platform-specific [type hints/typing] in a package's stub`
or `if a stub contains platform-specific types`

Author: cc @AlexWaygood as some who is great with wording :)

Reviewer 2: Maybe `if there is platform-specific code in a stubs package` or `if there are platform-specific branches in a stubs package`?
